definitions/features/candlepin_database.rb

```diff
@@ -27,12 +27,32 @@ def check_option_using_cpdb_help(option_name, parent_cmd = '')
 
   def execute_cpdb_validate_cmd
     main_cmd = cpdb_validate_cmd
-    unless main_cmd.empty?
-      main_cmd += format_shell_args(
-        '-u' => configuration['username'], '-p' => configuration[%(password)]
-      )
-      execute!(main_cmd, :hidden_patterns => [configuration['password']])
-    end
+    return [true, nil] if main_cmd.empty?
+    main_cmd += format_shell_args(
+      '-u' => configuration['username'], '-p' => configuration[%(password)]
+    )
+    execute_with_status(main_cmd, :hidden_patterns => [configuration['password']])
+  end
+
+  def db_backup_dir
```

Comment:
Since we are in the context of database class, the `db_` prefix is probably unnecessar